wireapp/wire-android-reloaded
# fix: new activity section scroll issue [AR-2279]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2279" title="AR-2279" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />AR-2279</a>  New Activity Section does not automatically appear without scrolling
  </td></table>
  <br />
 










----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When the user is on the conversation list, and a new message arrives, and “new activity” section should appear, then it’s hidden until the user scrolls up on the conversation list.

### Causes (Optional)

`LazyColumn` automatically tries to keep on screen previously visible items so it changes the scroll position when updating the order.

### Solutions

Add a dummy empty top item to keep the list not scrolled when updating and adding a new section.

When the list is scrolled to top and new items (e.g. new activity section) should appear on top of the list, it appears above all current items, scroll is preserved so the list still shows the same item as the first one on list so it scrolls automatically to that item and the newly added section on top is hidden above this previously top item, so for such situation when the list is scrolled to the top and we want the new section to appear at the top we need a dummy top item which will make it so it wants to keep this dummy top item as the first one on list and show all other items below it.

### Testing

#### How to Test

Open the app, don't scroll (keep the conversation list scrolled to top) and receive a message to make new activity section appear.

### Attachments (Optional)

https://user-images.githubusercontent.com/30429749/188395740-4c9defb0-ed71-4027-b1a9-2142331cb600.mp4

----
#### PR Post Submission Checklist for internal contributors (Optional)


----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
